### PR TITLE
fix(api): make api server configurable

### PR DESCRIPTION
**Pull request by ezreal1997**
[TRST-L1-M-6] Misconfigurations in HTTP server enables Slowloris and header overload attacks

issue: none

### CODE REVIEW

File: client/server/config.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	flags.StringVar(&cfg.APIAddress, "api-address", cfg.APIAddress, "The API server address to listen on")
+	flags.BoolVar(&cfg.EnableUnsafeCORS, "enabled-unsafe-cors", cfg.EnableUnsafeCORS, "Enable unsafe CORS for API server")
+	flags.UintVar(&cfg.ReadTimeout, "read-timeout", cfg.ReadTimeout, "Define the API server read timeout (in seconds)")
+	flags.UintVar(&cfg.ReadHeaderTimeout, "read-header-timeout", cfg.ReadHeaderTimeout, "Define the API server read header timeout (in seconds)")
+	flags.UintVar(&cfg.WriteTimeout, "write-timeout", cfg.WriteTimeout, "Define the API server write timeout (in seconds)")
+	flags.UintVar(&cfg.IdleTimeout, "idle-timeout", cfg.IdleTimeout, "Define the API server idle timeout (in seconds)")
+	flags.UintVar(&cfg.MaxHeaderBytes, "max-header-bytes", cfg.MaxHeaderBytes, "Define the API server max header (in bytes)")
+}
+
+type Config struct {
```
Comment (edisonz0718) [inline client/server/config.go:17]: How does the config load from the config file to Config?

Comment (ezreal1997) [reply client/server/config.go:17]: it's embedded in `story.toml` [config](https://github.com/piplabs/story/blob/rayden/api-server-config/client/config/config.go#L169)

Comment (ezreal1997) [reply client/server/config.go:17]: and will be read via https://github.com/piplabs/story/blob/rayden/api-server-config/lib/cmd/cmd.go#L74

**Review by edisonz0718 — APPROVED** [review]

**Comment by ezreal1997** [comment]
already tested:
1. run with previous `story.toml`, missed configs will be set to default value.
2. run with latest `story.toml` changes, update on configs working well.

cc @edisonz0718 @limengformal

**Review by 0xHansLee — APPROVED** [review]

**Review by jdubpark — APPROVED** [review]

**Pull request merged**